server/src/main/java/org/elasticsearch/cluster/health/ClusterIndexHealth.java

```diff
@@ -72,6 +76,11 @@ public ClusterIndexHealth(final IndexMetadata indexMetadata, final IndexRoutingT
             computeActiveShards += shardHealth.getActiveShards();
             computeRelocatingShards += shardHealth.getRelocatingShards();
             computeInitializingShards += shardHealth.getInitializingShards();
+
+            if (shardHealth.getUnassignedPrimaryShards() > 0) {
+                computeUnassignedPrimaryShards += shardHealth.getUnassignedPrimaryShards();
+            }
```

Comment:
I don't immediately see why we need this check? If the number of unassigned primary shards is 0, we can still just add it, right?